master/buildbot/www/avatar.py

```diff
@@ -51,6 +54,13 @@ def __init__(self,
         if github_api_endpoint is None:
             self.github_api_endpoint = self.DEFAULT_GITHUB_API_URL
         self.token = token
+        self.client_creds = None
```

Comment:
I think we should throw if only one of client_id and client_secret is specified and also if they are used together with token.

Reply:
Okay, I can do it.
But I don't really know how to add this to tests in a proper way.
Could you give me some guidance on this?

Reply:
The error is in configuration, so it's best to use `config.error` to raise it. As for testing, check out `with self.assertRaises(ConfigErrors):` in other tests. You will most likely just need to call the constructor in the `with` body.

Reply:
I tried something. Let me know if it fits.

Reply:
LGTM, thanks